From c72c2aae37118df1f8cf5cfad13a9aaef12c1c4e Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Mon, 26 Apr 2021 11:00:47 -0500
Subject: [PATCH 33/41] HSD #18016072895-1: firmware: stratix10-svc: add
 sessionID

commit 443890bcba11e10b629a9b3d8f6159e93c4049d3 from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.114-lts

Add SDM sessionID in PSGSIGMA_TEARDOWN command for FPGA attestation.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/firmware/stratix10-svc.c             | 2 +-
 include/linux/firmware/intel/stratix10-smc.h | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/firmware/stratix10-svc.c b/drivers/firmware/stratix10-svc.c
index 79cfb7fca8d8..aac358141a02 100644
--- a/drivers/firmware/stratix10-svc.c
+++ b/drivers/firmware/stratix10-svc.c
@@ -529,7 +529,7 @@ static int svc_normal_to_secure_thread(void *data)
 			break;
 		case COMMAND_FCS_PSGSIGMA_TEARDOWN:
 			a0 = INTEL_SIP_SMC_FCS_PSGSIGMA_TEARDOWN;
-			a1 = 0;
+			a1 = pdata->arg[0];
 			a2 = 0;
 			break;
 		case COMMAND_FCS_GET_CHIP_ID:
diff --git a/include/linux/firmware/intel/stratix10-smc.h b/include/linux/firmware/intel/stratix10-smc.h
index 544f46d03368..1398f37f0504 100644
--- a/include/linux/firmware/intel/stratix10-smc.h
+++ b/include/linux/firmware/intel/stratix10-smc.h
@@ -591,7 +591,8 @@ INTEL_SIP_SMC_FAST_CALL_VAL(INTEL_SIP_SMC_FUNCID_FPGA_CONFIG_COMPLETED_WRITE)
  *
  * Call register usage:
  * a0 INTEL_SIP_SMC_FCS_PSGSIGMA_TEARDOWN
- * a1-a7 not used
+ * a1 the session ID
+ * a2-a7 not used
  *
  * Return status:
  * a0 INTEL_SIP_SMC_STATUS_OK, INTEL_SIP_SMC_STATUS_ERROR or
-- 
2.18.1

